aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthService.kt

```diff
@@ -30,8 +30,8 @@ interface AWSCognitoAuthService {
 
                 CognitoIdentityProviderClient {
                     this.region = it.region
-                    this.endpointResolver = it.endpoint?.let { endpoint ->
-                        AWSEndpointResolver(Endpoint(endpoint))
+                    this.endpointProvider = it.endpoint?.let { endpoint ->
+                        AWSEndpointProvider(Endpoint(endpoint))
```

Comment:
```suggestion
                    this.endpointProvider = it.endpoint?.let { endpoint ->
                        EndpointProvider { Endpoint(endpoint) }
                    }
```

We can get rid of the below `AWSEndpointProvider` altogether.